lib/index.js

```diff
@@ -21,12 +21,16 @@ co(function * () {
 		cli.version(pkg)
 	} else {
 		var fly = yield cli.spawn(opts.file)
+		reporter.call(fly)
 
 		if (opts.list) {
 			cli.list(fly.host, {bare: opts.list === 'bare'})
 		} else {
-			return reporter.call(fly)
-				.emit('fly_run', {path: fly.file})
+			if (!fly.file) {
```

Comment:
You can move the `!fly.file` check ahead of the `if-else` check. The `list` option still requires a flyfile to be read & present.